tiledb/sm/subarray/subarray.cc

```diff
@@ -1552,7 +1539,10 @@ Status Subarray::compute_est_result_size(ThreadPool* const compute_tp) {
   if (est_result_size_computed_)
     return Status::Ok();
 
-  RETURN_NOT_OK(compute_tile_overlap(compute_tp));
+  // TODO: This routine is used in the path for the C APIs that estimate
+  // result sizes. We need to refactor this routine to handle the scenario
+  // where `tile_overlap_` may be truncated to fit the memory budget.
+  RETURN_NOT_OK(precompute_tile_overlap(0, range_num() - 1, compute_tp, true));
```

Comment:
We could explore the case where we can just throw an error if the full tile overlap size estimate exceeds `max_tile_overlap_size`. Also we can convert `max_tile_overlap_size` into a config parameter, or instead use the memory budget config parameter instead of that.